### ISSUE TITLE
How do I view Storm cluster data?

### ISSUE THREAD

**Issue by mydiemho** [linkedin/Burrow#39]
I see that we now have support for Storm offsets [https://github.com/linkedin/Burrow/pull/34] but there doesn't seem to be API for something like v2/storm [https://github.com/linkedin/Burrow/blob/master/http_server.go]

**Comment by sunh11373** [comment]
The storm offsets are now viewed the same way as Kafka offsets.  Though it could be improved to separate offsets of different types.

**Comment by mydiemho** [comment]
But when I access <INSTANCE>:8000/v2/kafka, I don't see the storm cluster.  This is what I have in config:

[storm "FOO"]
zookeeper=BAR1.com
zookeeper=BAR2.com
zookeeper=BAR3.com
zookeeper-port=2181
zookeeper-path=/kafka-spout

**Comment by toddpalino** [comment]
The problem here is that the storm cluster is not a separate cluster from the Kafka cluster, even though it's configured that way.

What we need is some documentation on setting up a storm cluster, because there's a requirement that you name the storm cluster the same as the Kafka cluster it maps to. However, I'm not sure that the config for the storm cluster should be in a separate section from the kafka cluster.

**Comment by mydiemho** [comment]
I see.  I was able to see the information after I added back the kafka cluster.  However, now I'm confused as to what zookeeper I should be using for each section since my kafka cluster and storm cluster have different zookeeper clusters.  There seems to be no difference in my testing though.  

[kafka "FOO"]
broker=kafka1.com
broker=kafka2.com
broker=kafka3.com
broker-port=9092
zookeeper=<??? SHOULD THIS BE THE CLUSTER USE BY KAFKA (kafkazk.com) or STORM (BAR1.com)>
zookeeper=<??? SHOULD THIS BE THE CLUSTER USE BY KAFKA OR STORM>
zookeeper=<??? SHOULD THIS BE THE CLUSTER USE BY KAFKA OR STORM>
zookeeper-port=2181
zookeeper-path=??

[storm "FOO"]
zookeeper=BAR1.com
zookeeper=BAR2.com
zookeeper=BAR3.com
zookeeper-port=2181
zookeeper-path=/kafka-spout

**Comment by hsun-cnnxty** [comment]
I think in [Kafka <cluster>] section, you should use the ZK instance that is used by Kafka cluster to store  metadata such as partitions, brokers etc. In [Storm <cluster>] section, you should use the ZK instance that is used by Storm Kafka Spout to store offsets.

**Comment by toff63** [comment]
@mydiemho did you manage to access storm consumer status?

@hsun-cnnxty @toddpalino I don't have much experience with Go and be completely wrong. 

But it looks like Storm client get partition offsets from Zookeeper and send then in the `offsetChannel`.

At the same time the http server is using the `requestChannel` to find Kafka consumers and their statuses. 

How is this suppose to work? Is there any way to get storm consumer status? I my analysis wrong or incomplete?

**Comment by hsun-cnnxty** [comment]
@toff63 "get storm consumer status", do you mean offsets?  In my environment, I can see the offsets with:

http://<burrow_endpoint>/v2/kafka/<cluster-name>/consumer/<spout_id>/topic/<topic_name>

and the result looks like:

```
{
  error: false,
  message: "consumer group topic offsets returned",
  offsets: [
    108617940,
    112588363,
    112110593
  ]
}
```

However, http://<burrow_endpoint>/v2/kafka/<cluster-name>/consumer/<spout_id>/status returns:

```
{
  error: false,
  message: "consumer group status returned",
  status: {
    cluster: "esg-stream",
    group: "RTDOrderTrackerTopology.KafkaSpout",
    status: "OK",
    complete: true,
    partitions: [ ]
  }
}
```

The missing of partition information may be limitation of current client. Seems it cannot find the list of topics in the ZK based on storm config.

-thanks

**Comment by toff63** [comment]
Thanks @hsun-cnnxty for your response. It confirmed I should be able to find storm consumer groups in the list of Kafka consumer groups at `http://<burrow_endpoint>/v2/kafka/<cluster-name>/consumer`. 

Investigating a bit more I found a bug preventing my storm consumer groups from appearing. I submitted a PR to fix it. https://github.com/linkedin/Burrow/pull/51

**Comment by toddpalino** [comment]
Merged in #51. Thanks for the help, @toff63.
